nixos/tests/irqbalance.nix

```diff
@@ -0,0 +1,31 @@
+{ pkgs, lib, ... }:
+{
+  name = "irqbalance";
+  meta.maintainers = with lib.maintainers; [ h7x4 ];
+
+  nodes.machine =
+    { config, ... }:
+    {
+      virtualisation.cores = 2;
+      services.irqbalance.enable = true;
+
+      systemd.services.irqbalance.serviceConfig.ExecStart = [
+        ""
+        "${lib.getExe config.services.irqbalance.package} --journal --debug"
+      ];
+    };
+
+  testScript = ''
+    machine.wait_for_unit("irqbalance.service")
+
+    machine.wait_until_succeeds("journalctl -u irqbalance.service --grep='Package 0'")
+
+    machine.sleep(5)
+
+    unmanaged_irq_count = machine.succeed("journalctl -u irqbalance.service -o cat --grep 'affinity is now unmanaged' | sort -u | wc -l")
```

Comment:
I'd argue that it'd make more sense to use `machine.wait_until_succeeds` here. `sleep` is super error-prone, especially on build infra.

Reply:
I don't mind changing it to `wait_until_succeeds`, but I reckon this command will always succeed and provide either of the numbers 0, 1 or 2? I suppose I could remove the `sleep` maybe, since we're already waiting for `Package 0` beforehand.

What do you think?

Reply:
>  I suppose I could remove the sleep maybe, since we're already waiting for Package 0 beforehand

If it's stable without the sleep, then go for it.
But `sleep` + `machine.succeed()` after that is always a red flag (and usually an indication that you want `wait_until_succeeds`).